src/aleph/http.clj

```diff
@@ -100,20 +100,23 @@
    will be errors, and a new connection must be created."
   [^URI uri options middleware on-closed]
   (let [scheme (.getScheme uri)
-        ssl? (= "https" scheme)]
-    (-> (client/http-connection
-          (InetSocketAddress/createUnresolved
-            (.getHost uri)
-            (int
-              (or
-                (when (pos? (.getPort uri)) (.getPort uri))
-                (if ssl? 443 80))))
-          ssl?
-          (if on-closed
-            (assoc options :on-closed on-closed)
-            options))
-
-        (d/chain' middleware))))
+        ssl? (= "https" scheme)
+        conn (client/http-connection
+              (InetSocketAddress/createUnresolved
+               (.getHost uri)
+               (int
+                (or
+                 (when (pos? (.getPort uri)) (.getPort uri))
+                 (if ssl? 443 80))))
+              ssl?
+              (if on-closed
+                (assoc options :on-closed on-closed)
+                options))]
+    (doto (d/chain' conn middleware)
+      (d/catch' (fn [e]
+                  (log/trace e "Terminating creation of HTTP connection")
+                  (d/error! conn e)
+                  (d/error-deferred e))))))
```

Comment:
Can you explain this to me? It looks like you're using `doto` to return the deferred from `chain'`, but then you're returning an error-deferred from the `catch'` that will go unused, afaict.

If you don't care about the return value, and just want side effects, it may be clearer to replace the `catch'` with `on-realized`

Reply:
Oof, you're right, `on-realized` would make much more sense here to achieve the desired result, will adjust accordingly.

 The `error-deferred` return is in fact a leftover of an older version, thanks for noticing! Will also clean that up.

Reply:
OK replaced all instances of this with `(d/on-realized x identity ...)` instead and dropped the pointless `error-deferred`.